src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -505,19 +505,19 @@ def inner(*args, **kwargs):
                 content_length = int(cherrypy.request.headers['Content-Length'])
                 body = cherrypy.request.body.read(content_length)
                 if not body:
-                    return func(*args, **kwargs)
-
-                try:
-                    data = json.loads(body.decode('utf-8'))
-                except Exception as e:
-                    raise cherrypy.HTTPError(400, 'Failed to decode JSON: {}'
-                                             .format(str(e)))
-                kwargs.update(data.items())
-                ret = func(*args, **kwargs)
+                    ret = func(*args, **kwargs)
```

Comment:
As these changes are only visual and aren't required to resolve the issue, you may want to consider to separate all changes of this file into a separate commit.

Personally, I like how the code was structured before.

Reply:
These changes are not only cosmetic, it's a bug fix. You may notice that the code path has been changed significantly. The old code does not convert the result content into JSON for some paths which is totally wrong.